### PR TITLE
Annotations for user_erasure_store

### PR DESCRIPTION
**Pull request by DMRobertson**
I'm not sure why this was excluded---it seemed to be passing for me. But
it's easy enough to fixup.

### CODE REVIEW

File: synapse/storage/databases/main/user_erasure_store.py [hunk 1]
```diff
@@ -14,11 +14,12 @@
...
 from typing import Dict, Iterable
 
-from synapse.storage._base import SQLBaseStore
+from synapse.storage.database import LoggingTransaction
+from synapse.storage.databases.main import CacheInvalidationWorkerStore
 from synapse.util.caches.descriptors import cached, cachedList
 
 
-class UserErasureWorkerStore(SQLBaseStore):
+class UserErasureWorkerStore(CacheInvalidationWorkerStore):
```
Comment (clokep) [inline synapse/storage/databases/main/user_erasure_store.py:22]: I'm not sure about this change, is this due to it calling methods from that?

Comment (DMRobertson) [reply synapse/storage/databases/main/user_erasure_store.py:22]: Yes, `_invalidate_cache_and_stream` is the method in question. Sorry, should have written that down.

**Review by clokep — APPROVED** [review]
I think this should be OK.

**Comment by DMRobertson** [comment]
Merge commit to resolve mypy.ini.

**Pull request merged**